### PR TITLE
feat: deploy to BOB, Mantle, Taiko

### PR DESCRIPTION
**Pull request by paulbalaji**
resolves https://github.com/hyperlane-xyz/issues/issues/1289

- feat: deploy to BOB, Mantle, Taiko
- drive-by: fix deploying to specific chain
- drive-by: adding AVS operators for eth
- drive-by: support non-evm remotes in mainnet3 core config

### CODE REVIEW

**Review by tkporter — APPROVED** [review]
lgtm - some questions on overrides

also does this fix https://github.com/hyperlane-xyz/hyperlane-monorepo/issues/4087? or is there more to do there

File: typescript/infra/config/environments/mainnet3/chains.ts [hunk 1]
```diff
@@ -29,6 +34,11 @@ export const chainMetadataOverrides: ChainMap<Partial<ChainMetadata>> = {
       maxPriorityFeePerGas: 50 * 10 ** 9, // 50 gwei
     },
   },
+  polygonzkevm: {
+    transactionOverrides: {
+      gasPrice: 2 * 10 ** 8, // 0.2 gwei
```
[the remark this answers is not in the document]
Comment (paulbalaji) [reply typescript/infra/config/environments/mainnet3/chains.ts:39]: So I set this mainly for contract deployment sake and not for the agents, I found the default gas estimation for zkevm specifically to result in a lot of flakes - why is one of the main reasons we ended up having to run deploy scripts multiple times

setting this eliminated flakiness of deploying core almost entirely

File: typescript/infra/config/environments/mainnet3/chains.ts [hunk 2]
```diff
@@ -18,6 +18,11 @@ export const chainMetadataOverrides: ChainMap<Partial<ChainMetadata>> = {
       gasPrice: 3 * 10 ** 9, // 3 gwei
     },
   },
+  mantle: {
+    blocks: {
+      confirmations: 3,
```
[the remark this answers is not in the document]
Comment (paulbalaji) [reply typescript/infra/config/environments/mainnet3/chains.ts:23]: same as zkevm below, as per the spec and what I can observe from the explorers - the original metadata of requiring 1 confirmation seems correct

however, when doing our deployments I regularly ran into nonce reuse errors despite it being OK on the explorer - increasing the #confirmations required added some buffer here to not see those errors but also be able to deploy

File: typescript/infra/src/deployment/deploy.ts [hunk 3]
```diff
@@ -83,7 +83,7 @@ export async function deployWithArtifacts<Config extends object>({
   try {
     await deployer.deploy(targetConfigMap);
   } catch (e) {
-    console.error('Failed to deploy contracts', e);
+    console.warn('Contract deployment exceeding configured timeout', e);
```
[the remark this answers is not in the document]
Comment (paulbalaji) [reply typescript/infra/src/deployment/deploy.ts:86]: True - will update the message so it warns on timeouts and logs errors for everything else

File: typescript/infra/config/environments/mainnet3/chains.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (paulbalaji) [reply typescript/infra/config/environments/mainnet3/chains.ts:23]: moving to registry https://github.com/hyperlane-xyz/hyperlane-registry/pull/82

File: typescript/infra/config/environments/mainnet3/chains.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (paulbalaji) [reply typescript/infra/config/environments/mainnet3/chains.ts:39]: will update to 1 gwei in overrides

**Pull request merged**